### PR TITLE
Epic 7 - Auth via auth channel

### LINKED ISSUE #1412: refactor new auth token message into its own channel
auth tokens are currently listened to by notifications, or maybe some other mutation or subscription.
Instead, we should just dedicate a channel to always listen to a new auth token. that way, the pattern remains the same whether i'm doing it myself (addOrg) or someone else does it to me (kick out of org).

### PR DESCRIPTION
**Pull request by mattkrick**
fixes #1412 auth token channel
#1391

### CODE REVIEW

**Comment by mattkrick** [comment]
TESTS

Auth token 
- [ ] User 2 leaves a team. User 1 invites them back, they can immediately access that team
- [ ] User 2 gets invited to a new team. they accept via notification. they can immediately access that team
- [ ] user 1 creates a new org & team. they can immediately see the team dash
- [ ] user 2 gets kicked off a team while they are on the team dash. they can no longer see the team dash, even if they hit the back button

Organizations
- [ ] If an org has 1 person on it, it says "1 User" if it has more, it says "Users"

OrgBilling
- [ ] for a personal org with 1 member, they click how much will it cost me. it says 1 User. For larger orgs, it says "Users"
- [ ] Upgrade to Pro from Org Billing. add credit card "4242424242424242", expiry "12/23" or something in the future. it works 
- [ ] Once Pro, you see all the pro badges
- [ ] you see a danger zone. clicking "contact us" opens up an email to love@parabol.co so they can easily unsubscribe

**Comment by mattkrick** [comment]
@jordanh a couple more tests for ya if you've got some time

File: src/server/graphql/models/Team/archiveTeam/__tests__/archiveTeam.test.js [hunk 1]
```diff
@@ -62,6 +63,7 @@ describe('ArchiveTeam', () => {
     auth0ManagementClient.users.updateAppMetadata.mockReset();
     exchange.publish.mockReset();
     getPubSub.default = () => mockPubSub;
+    tmsSignToken.default = jest.fn(() => 'FAKEENCODEDJWT');
```
Comment (dan-f) [inline src/server/graphql/models/Team/archiveTeam/__tests__/archiveTeam.test.js]: Looks like you can just do this once at the top of the file, no?

Comment (mattkrick) [reply src/server/graphql/models/Team/archiveTeam/__tests__/archiveTeam.test.js]: concurrently run tests make for sad times if i do that :-(

**Comment by dan-f** [comment]
@mattkrick I get this gnarly screen when I try to accept an invite from my teammate:

screen shot 2017-10-24 at 3 53 45 pm

**Comment by mattkrick** [comment]
oh well that is super! I probably shouldn't print errors to the user. more harm than good.

**Comment by mattkrick** [comment]
hey @dan-f could you give me some steps to reproduce? it seems to be working OK for me.

**Comment by dan-f** [comment]
Sure, let me also rule out the likely possibility that I could have missed a few steps in setting up my env after changing branches. Anything I need to do other than:

- `yarn`
- `yarn db:migrate`
- `yarn build:relay`
- `yarn build:dll`?

As for steps, I believe I:

- had two users: A and B
- A invites B to a team
- B navigates to B's notifications
- B clicks "accept"
- No feedback given to user B
- B clicks "accept" again
- B sees error page

**Comment by mattkrick** [comment]
that no feedback given to user B is interesting, cuz it should be removed optimistically. can you repro during a share screen?

**Comment by dan-f** [comment]
@mattkrick here's my pass at the tests. A few cases failed for me:

Auth token 
- [x] User 2 leaves a team. User 1 invites them back, they can immediately access that team
- [x] User 2 gets invited to a new team. they accept via notification. they can immediately access that team
- [x] user 1 creates a new org & team. they can immediately see the team dash
- [ ] ❌ user 2 gets kicked off a team while they are on the team dash. they can no longer see the team dash, even if they hit the back button
    - user 2 can still see the team that they were kicked off of. back/forward work as if they hadn't been kicked off of the team

Organizations
- [ ] If an org has 1 person on it, it says "1 User" if it has more, it says "Users"

OrgBilling
- [x] for a personal org with 1 member, they click how much will it cost me. it says 1 User. For larger orgs, it says "Users"
- [ ] ❌  Upgrade to Pro from Org Billing. add credit card "4242424242424242", expiry "12/23" or something in the future. it works 
    - can't test this one. not sure if there's a magic CVC, but on submit I see "Invalid API Key provided"
- [ ] Once Pro, you see all the pro badges
- [ ] you see a danger zone. clicking "contact us" opens up an email to love@parabol.co so they can easily unsubscribe

**Comment by mattkrick** [comment]
good stuff! that first error is OK< cashay just aggressively caches, we can fix this when relay happens. the important thing is that they can't see any _new_ information.

the 2nd one is a fun one, i must not have put the API key on the dev server, i'll make sure it's there & in the mean time, i think we can push that to staging

**Pull request merged**